agent/structs/acl.go

```diff
@@ -261,6 +262,35 @@ type ACLToken struct {
 	RaftIndex
 }
 
+func (t *ACLToken) UnmarshalJSON(data []byte) (err error) {
+	type Alias ACLToken
+	aux := &struct {
+		ExpirationTTL interface{}
+		Hash          string
+		*Alias
+	}{
+		Alias: (*Alias)(t),
+	}
+	if err = json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+	if aux.ExpirationTTL != nil {
+		switch v := aux.ExpirationTTL.(type) {
+		case string:
+			if t.ExpirationTTL, err = time.ParseDuration(v); err != nil {
+				return err
+			}
+		case float64:
+			t.ExpirationTTL = time.Duration(v)
+		}
+
+	}
+	if aux.Hash != "" {
+		t.Hash = []byte(aux.Hash)
```

Comment:
Should we base64 decode this?

Reply:
Sounds good. Maybe we should revisit this in the future under a different task of figuring out if the current decoding behavior is correct. I am fairly certain that it gets base64 encoded on the way out but encoding/json during the decode doesn't attempt to base64 decode it.

Another possible solution is to ignore the Hash coming from the request and calculate the correct one (that will be done for any write request in the RPC API anyways so its probably not important.